psl-clj/pom.xml

```diff
@@ -0,0 +1,172 @@
+    -
+    - Licensed under the Apache License, Version 2.0 (the "License");
+    - you may not use this file except in compliance with the License.
+    - You may obtain a copy of the License at
+    -
+    - http://www.apache.org/licenses/LICENSE-2.0
+    -
+    - Unless required by applicable law or agreed to in writing, software
+    - distributed under the License is distributed on an "AS IS" BASIS,
+    - WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    - See the License for the specific language governing permissions and
+    - limitations under the License.
+  -->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	 xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>org.linqs</groupId>
+  <artifactId>psl-clj</artifactId>
+  <packaging>jar</packaging>
+  <name>psl-clj</name>
+  <description>A clojure user interface for PSL</description>
+  <parent>
+    <groupId>org.linqs</groupId>
+    <artifactId>psl-experimental</artifactId>
+    <version>2.1.0-SNAPSHOT</version>
+  </parent>
+  <distributionManagement>
+    <repository>
+      <id>clojars</id>
+      <name>Clojars repository</name>
+      <url>https://clojars.org/repo</url>
+    </repository>
+  </distributionManagement>
+  <build>
+    <sourceDirectory>src</sourceDirectory>
+    <testSourceDirectory>test</testSourceDirectory>
+    <resources>
+      <resource>
+        <directory>resources</directory>
+      </resource>
+    </resources>
+    <testResources>
+      <testResource>
+        <directory>resources</directory>
+      </testResource>
+    </testResources>
+    <directory>target</directory>
+    <outputDirectory>target/classes</outputDirectory>
+    <plugins>
+      <plugin>
+        <groupId>com.theoryinpractise</groupId>
+        <artifactId>clojure-maven-plugin</artifactId>
+        <version>1.8.1</version>
+        <extensions>true</extensions>
+        <configuration>
+	  <sourceDirectories>
+	    <sourceDirectory>src</sourceDirectory>
+	  </sourceDirectories>
+        </configuration>
+        <executions>
+	  <execution>
+	    <id>compile-clojure</id>
+	    <phase>compile</phase>
+	    <goals>
+	      <goal>compile</goal>
+	    </goals>
+	  </execution>
+	</executions>
+      </plugin>
+    </plugins>
+  </build>
+  <repositories>
+    <repository>
+      <id>clojars</id>
+      <url>https://clojars.org/repo/</url>
+      <snapshots>
+        <enabled>true</enabled>
+      </snapshots>
+      <releases>
+        <enabled>true</enabled>
+      </releases>
+    </repository>
+  </repositories>
+  <dependencies>
+    <dependency>
+      <groupId>org.linqs</groupId>
+      <artifactId>psl-core</artifactId>
+      <version>${psl-core-version}</version>
+    </dependency>
+    <dependency>
+      <groupId>org.linqs</groupId>
+      <artifactId>psl-groovy</artifactId>
```

Comment:
Do you actually need the groovy functionality, or can you call directly into core?
I think what we should do is either have this go off of core directly, or have it work off of the java interface (which doesn't exist yet).
I assume you are using the groovy interface for the add() method?

Reply:
Correct, it uses the add() method.  The last time I checked, core did not include all that was needed, but that may well have changed with all the recent improvements.